### PR TITLE
feat(button): Implemented the button component

### LINKED ISSUE #8: Button component
#### Screenshots
Screenshot 2020-03-14 at 23 17 27
Screenshot 2020-03-14 at 23 17 23

-----

#### Used by
 - All the apps

#### Names
 - Button

#### Props
 - label (`string`)
 - onClick (`function`)
 - type (`enum` `{ primary | secondary }`)

#### Other
 - We also have buttons for login form, should we have the same component for those as well?

### PR DESCRIPTION
**Pull request by surdu**
Closes #8

Implements the Button component

### CODE REVIEW

File: src/components/button/button.js [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
+      {children}
+    </button>
+  );
 };
 
 Button.propTypes = {
-  children: PropTypes.node.isRequired
+  children: PropTypes.node.isRequired,
+  onClick: PropTypes.func.isRequired,
+  type: PropTypes.oneOf(["primary", "danger"]),
```
Comment (moonflare) [inline src/components/button/button.js]: We could set `type: primary` as default. Please check [this](https://reactjs.org/docs/typechecking-with-proptypes.html#default-prop-values) article, same for `inverted`

File: src/components/button/button.js [hunk 2]
```diff
@@ -1,10 +1,30 @@
...
+  }
+
+  if (inverted) {
+    additionalClasses.push("is-outlined");
+  }
+
+  return (
+    <button
+      onClick={onClick}
+      className={"button is-primary " + additionalClasses.join(" ")}
```
Comment (moonflare) [inline src/components/button/button.js]: We should not hardcode `is-primary` here. It's true it's gonna be overriden by the other types, but in this case we won't use the styles from it.
```suggestion
      className={"button " + additionalClasses.join(" ")}
```

File: src/styles.scss [hunk 3]
```diff
@@ -1,3 +1,6 @@
+$primary: #1C1080;
+$danger: #FF1616;
```
Comment (moonflare) [inline src/styles.scss]: We will need to change some of the CSS for the button, like some of the below. Can you please check Figma for the proper styles?
```css
font-weight: bold;
font-size: 14px;
line-height: 21px;
border-radius: 3px;
```

File: src/components/button/button.js (hunk 2 above)
Comment (moonflare) [reply src/components/button/button.js]: Also, for dynamic classes we could use [classNames](https://github.com/JedWatson/classnames#usage), you have an example on this [wip branch](https://github.com/code4romania/taskforce-fe-components/blob/dan/%234_add-list-components/src/components/ListItem/ListItem.js#L10)

File: src/styles.scss (hunk 3 above)
Comment (surdu) [reply src/styles.scss]: Sure thing!

File: src/components/button/button.js (hunk 2 above)
Comment (surdu) [reply src/components/button/button.js]: That `is-primary` there was forgotten, my bad.

As for the classNames, I simplified the code for that. Please advise if you still consider adding an extra package for this looks necessary.

File: src/styles.scss (hunk 3 above)
Comment (surdu) [reply src/styles.scss]: Font size is all over the place in Figma. I went with 16px, seemed like the median :)

File: src/components/button/button.scss [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.button {
+  font-family: Titillium Web;
```
Comment (moonflare) [inline src/components/button/button.scss]: We won't need this one here. It's been cover in this PR https://github.com/code4romania/taskforce-fe-components/pull/21

Same for the `.storybook/preview-head.html` file
```suggestion
```

**Comment by moonflare** [comment]
@surdu If you could also check the merge conflict please? 😅

**Comment by surdu** [comment]
Done & done

**Pull request merged**